tests/php/ORM/DataObjectTest.php

```diff
@@ -826,10 +826,6 @@ public function testHasOneAsField()
         $team1->Captain = $captain2;
         $team1->write();
         $this->assertEquals($captain2->ID, $team1->Captain->ID);
-
-        // Setter: Custom data (required by DataDifferencer)
```

Comment:
I looked into this and experimented with some test objects on my local with $has_one relationships and putting them into DataDifferencer - as far as I can tell this commet about it being required by DataDifferencer is incorrect.  The existing DataDifferenceTest still passes its unit tests and does not throw deprecation warnings.

I deleted this unit test because it end up trying to create a dynamic property called Captain in (from memory)  DBHTMLText